dd-sdk-android/src/main/kotlin/com/datadog/android/rum/gestures/DatadogGesturesListener.kt

```diff
@@ -0,0 +1,149 @@
+                    e.y
+                )
+            spanBuilder.withTag(TAG_TARGET_CLASS_NAME, target.javaClass.canonicalName)
+            // we will handle general exceptions here. We don't know what might go wrong apart from
+            // ResourceNotFound exception.
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                spanBuilder.withTag(
+                    TAG_TARGET_RESOURCE_ID,
+                    target.resources.getResourceEntryName(target.id)
+                )
+            } catch (e: Exception) {
+                sdkLogger.e("$TAG: Could not found resource name for target:${target.id}", e)
+            }
+            spanBuilder
+                .start()
+                .finish(DEFAULT_EVENT_DURATION)
+        }
+        return false
+    }
+
+    override fun onDown(e: MotionEvent): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onFling(
+        e1: MotionEvent,
+        e2: MotionEvent,
+        velocityX: Float,
+        velocityY: Float
+    ): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onScroll(
+        e1: MotionEvent,
+        e2: MotionEvent,
+        distanceX: Float,
+        distanceY: Float
+    ): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onLongPress(e: MotionEvent) {
+        // No Op
+    }
+
+    // endregion
+
+    companion object {
+        private const val TAG = "DatadogGestureListener"
+        internal const val UI_TAP_ACTION_EVENT = "TapEvent"
+        internal const val TAG_TARGET_CLASS_NAME = "target.classname"
+        internal const val TAG_TARGET_RESOURCE_ID = "target.resourceId"
+        val DEFAULT_EVENT_DURATION = TimeUnit.MILLISECONDS.toMicros(16)
+
+        private fun findTarget(decorView: View, x: Float, y: Float): View {
+            val stack = LinkedList<View>()
+            stack.addFirst(decorView)
+            while (stack.isNotEmpty()) {
+                val view = stack.removeFirst()
+
+                if (isValidTarget(view)) {
+                    return view
+                }
+
+                if (view is ViewGroup) {
+                    handleViewGroup(view, x, y, stack)
+                }
+            }
+
+            return decorView
+        }
+
+        private fun handleViewGroup(
+            view: ViewGroup,
+            x: Float,
+            y: Float,
+            stack: LinkedList<View>
+        ) {
+            for (i in 0 until view.childCount) {
+                val child = view.getChildAt(i)
+                if (hitTest(child, x, y)) {
+                    stack.add(child)
+                }
+            }
+        }
+
+        private fun isValidTarget(view: View): Boolean {
+            if (!(view.isClickable && view.visibility == View.VISIBLE)) {
```

Comment:
Invisible views are clickable too; and can provide useful info in the span

Reply:
Do you guys think we should add invisible targets ? What will happen if we have an "invisible" on top of a "visible" one. The way we have the code now the "invisible" may steal the target to the "visible" one. I guess for these cases we will need to order to "children" by z-index, visibility and meaning another complexity added to the code there. Do you think it makes sense to add this now ?

Reply:
No opinion. Best may be to have a way to report both (with a tag/log?)

Reply:
I would rather keep it light and simple for now. If we see we need this we can add it later.